### ISSUE TITLE
Save attachments to Message model

### ISSUE THREAD

**Issue by aaronschachter** [DoSomething/gambit-conversations#31]
We'll need to store any attachments that exist for inbound and outbound messages.

For outbound messages, we currently don't have any fields in Contentful to use for attachments on an outbound template. However, an agent could include attachments via Front that we'd want to save to our `Message` model.

**Comment by rapala61** [comment]
I can take on this. How does that attachment URL gets sent to the user? As part of the text response?

I need more info to get to work on this as I'm not familiar with how Front includes this URL. This looks like an specific parsing middleware/function for incoming Front messages, but it seems related to the incoming messages with mms from Twilio too. Should this functionality be included in the middleware we are thinking about implementing to deprecate `Adapters`? Ultimately, the `Message`  will be updated with an `attachment` property, but how to get there needs a little more fleshing out.

**Comment by aaronschachter** [comment]
When thinking through outbound attachments, this issue turns into a really big one which would also mean a really big PR if it was in one branch (e.g. parsing Front)

@rapala61 Maybe let's just start with storing inbound messages attachments from Twilio messages? Which means this is blocked by https://github.com/DoSomething/gambit-conversations/pull/50/files#r132314865

**Comment by rapala61** [comment]
This task addresses just the attachment problem, but I know we want to port the whole of the functionality from Adapters into Convo API, should this task (#31) be included in the bigger [Adapters -> Convo API migration](https://github.com/DoSomething/gambit-conversations/issues/53)? The many similar issues are a little confusing to scope separately @aaronschachter

**Comment by aaronschachter** [comment]
this issue was created to document that we need both inbound + outbound messages.

I think it makes sense to add storing inbound messages along with #31 but outbound should be a separate PR to making reviewing easier. I'm open to renaming this issue as inbound message storage and opening an outbound message storage @rapala61 if that would be easier...

**Comment by rapala61** [comment]
@aaronschachter I think renaming and creating a separate for outbound messages is a good idea.

**Comment by rapala61** [comment]
@aaronschachter 
This is an example of an incoming payload we should get from Blink (redacted for privacy):
```
{
  "ToCountry": "US",
  "MediaContentType0": "image/png",
  "ToState": "",
  "SmsMessageSid": "MM09a8f657567f807443191c1e7exxxxxx",
  "NumMedia": "1",
  "ToCity": "",
  "FromZip": "10010",
  "SmsSid": "MM09a8f657567f807443191c1e7exxxxxx",
  "FromState": "NY",
  "SmsStatus": "received",
  "FromCity": "NEW YORK",
  "Body": "",
  "FromCountry": "US",
  "To": "xxxxx",
  "ToZip": "",
  "NumSegments": "1",
  "MessageSid": "MM09a8f657567f807443191c1e7exxxxxx",
  "From": "+slothielandphone#",
  "MediaUrl0": "https://api.twilio.com/2010-04-01/Accounts/ACd3386eff2c87ebd2f00b628e02xxxxxx/Messages/MM09a8f657567f807443191c1e7exxxxxx/Media/ME0bdfda835a49390c85311e6d27xxxxxx",
  "ApiVersion": "2010-04-01"
}
```
I think we should store more than the url (`MediaUrl0`), maybe also the content type (`MediaContentType0`). The attachments array in the message instance would contain objects with `url` and `contentType` properties:
```
{
    attachments: [
        {
            url: 'https://api.twilio.com/2010-04-01/Accounts/ACd3386eff2c87ebd2f00b628e02xxxxxx/Messages/MM09a8f657567f807443191c1e7exxxxxx/Media/ME0bdfda835a49390c85311e6d27xxxxxx',
            contentType: 'image/png'
        }
    ]
}
```

**Comment by aaronschachter** [comment]
Sounds good!

**Comment by aaronschachter** [comment]
I've started a [Wishlist wiki page](https://github.com/dosomething/gambit-conversations/wiki) to document nice-to-have features like storing Outbound message attachment.
